File: src/1304.find-n-unique-integers-sum-up-to-zero.1426/solution.js
/**
 * @param {number} n
 * @return {number[]}
 */
var sumZero = function (n) {
    const result = [];
    // 奇数的多加个0就好了
    if (n & 1) {
        result.push(0);
        n--;
    }
    // 剩下的就是正负对应着来
    n /= 2;
    for (let i = 1; i <= n; i++) {
        result.push(i);
        result.push(-i);
    }
    return result;
};

File: src/1304.find-n-unique-integers-sum-up-to-zero.1426/solution.java
class Solution {
    public int[] sumZero(int n) {
        int[] result = new int[n];
        for(int i=0;i<n/2;i++){
            result[2*i] = i+1;
            result[2*i+1] = -(i+1);
        }
        return result;
    }
}